Feat/mesto update

sergey-kouzma opened the pull request.

sergey-kouzma committed:
Merge pull request #1 from sergey-kouzma/feat/mesto-update-api

api

Vasyankin-Alexander-Henchfiend committed:
валидация по ООП ч.1

- добавлен класс FormValidation на универсальный инструмент для валидации любой формы.
- принцип его работы - нужно дать ему параметры настроек (config - объект с нужными селекторами элементов формы: инпуты, поля для ошибки, кнопки, сама форма) и непосредственно модальное окно которое будет валидироваться (любое с инпутами и селектаровами как в config), получив эти данные класс подключается ко всем полям воода и контролирует их заполнение, показывая сообщения о неверном заполнении и активируя/деактивируя кнопку
- запуск валидации после создания класса производится с помощью метода enableValidation
- класс импортирован в index.js и подключен, пока только для окна редактирования профиля
- создан и также подключен отдельный файл, содержащими параметры для класса
- произвести тестирование не удалось, так как не получилось запустить DEV сервер у себя, причину обнаружить не смог.

Vasyankin-Alexander-Henchfiend committed:
Валидация на ООП ч2

- установлен webpack, dev сервер запускается
- сборка протестирована работает
- проведен рефакторинг, const errorElement выведен в метод _setEventListeners, чтобы не дублировать его объявление в методах _showError и _hideError
- изменен способ поиска поля ввода для отображения ошибки
- используемые параметры config объявлены отдельно
- созраняется баг с отображение подсказки об обишке в том, случае если попап был закрыт без сохранения данных

Vasyankin-Alexander-Henchfiend committed:
валидация на ООП ч3

- для вызова класса валиции теперь используется перебор массива всех форм на странице

Vasyankin-Alexander-Henchfiend committed:
валидация ч4

- доработана форма добавления аватара, чтобы кнопка оставалась не активной после первого добавления картинки

Vasyankin-Alexander-Henchfiend committed:
Валидация на ООП, итерация 2

- учтены замечания на ревью

sergey-kouzma committed:
Merge pull request #3 from sergey-kouzma/feat/mesto-update-validation

Feat/mesto update validation

Vasyankin-Alexander-Henchfiend committed:
Класс UserInfo ч1

- класс UserInfo создан
- его метод setUserInfo подключен к Api и функциям смены информации о профиле(имя, описание, аватар)
- создана отдельная функция для смены названия кнопки сохранения во всех формах
- константы касающиеся работы класса UserInfo выведены в отдельный модуль
- поправлены стили аватара и картинок карточек, чтобы изображение не ломалось, когда загружаются картинки с разними соотношениями сторон
- сборка протестирована на локальном сервере

sergey-kouzma committed:
card

sergey-kouzma committed:
Merge pull request #5 from sergey-kouzma/feat/mesto-update-card

Feat/mesto update card

Vasyankin-Alexander-Henchfiend committed:
класс UserInfo

- налажена работа класса с функциями смены аватара и профиля
- создан метод getUserInfo
- небольшой рефакторинг (убраны закоментированные строки, функции переделаны в стрелочные)

Vasyankin-Alexander-Henchfiend committed:
класс UserInfo ч3

- метод getUserInfo подключен
- сборка протестирована

sergey-kouzma committed:
Merge pull request #6 from sergey-kouzma/feat/mesto-update-profile

Feat/mesto update profile

sergey-kouzma committed:
popup

Vasyankin-Alexander-Henchfiend committed:
класс Section ч1

- класс Section создан в соответствии с тех.заданием

Vasyankin-Alexander-Henchfiend committed:
класс Section ч2

- продолжаем попытки подключить класс Section к работе

Vasyankin-Alexander-Henchfiend committed:
класс Section ч3

- подгрузка картинок с сервера реализована через класс Section
- поломана функция добавления новых картинок через форму картинок

Vasyankin-Alexander-Henchfiend committed:
класс Section ч4

- удалены закоменнтированные строки

sergey-kouzma committed:
check

sergey-kouzma committed:
Merge pull request #7 from sergey-kouzma/feat/mesto-update-section

Feat/mesto update section

Vasyankin-Alexander-Henchfiend committed:
рефакторинг класса Card ч1

- создание Template выведено в отдельный метод
- уменьшино количество аргументов которые принимает класс Card, что позитивно сказывается на его вызове

sergey-kouzma committed:
conflicts

sergey-kouzma committed:
merge

Vasyankin-Alexander-Henchfiend committed:
рефакторинг класса Card ч2

- класс Card теперь принимает всего два параметра, что облегчает его вызов
- введены отдельные методы, для определения загружена ли карточка с нашей учетной записи(появляется корзина), лайкали ли мы её(меняем иконку лайка)
- общий рефакторинг(уделены лишние скобки, изменен порядок строк, чтобы было проще воспринимать)

Vasyankin-Alexander-Henchfiend committed:
рефакторинг формы добавления карточки

- удален закоментированный код
- добавлена функция смены надписи на кнопке сохранения
- предпримята попытка подгружать новую карточку через ту же функцию, что и загружаем карточки с сервера -- неудачно, так как класс Section принимает массив, а не объект.

Vasyankin-Alexander-Henchfiend committed:
функция добавления новой карточки

- функция реализована через вызов классов Section и Card
- не используемые функции и переменные удалены

to-review committed:
формы работают

to-review committed:
работа аватара

to-review committed:
работа формы добавления метса

to-review committed:
работа попапа картинки

sergey-kouzma committed:
Merge pull request #9 from sergey-kouzma/feat/mesto-update-popup

Feat/mesto update popup

Vasyankin-Alexander-Henchfiend committed:
Merge branch 'feat/mesto-update-card-update-1' into feat/mesto-update

Vasyankin-Alexander-Henchfiend committed:
Merge pull request #11 from sergey-kouzma/feat/mesto-update

Feat/mesto update

gennady-bars reviewed (commented):
-------------------------


 в процессе ревью

Vasyankin-Alexander-Henchfiend committed:
решение конфликтов исправление ошибок

- исправлены ошибки полученные при неправильном решении конфликтов в ветках
- удалены лишние файлы, закомментированные строки, объявленные но не вызванные переменные
- сборка протестирована

gennady-bars reviewed (commented):
К сожалению, работу придется отклонить без потери итерации, так как проект не запускается. Скрин ошибки https://disk.yandex.ru/i/aIfhVN-6jgK2dg

sergey-kouzma committed:
Merge pull request #12 from sergey-kouzma/feat/mesto-update-card-update-1

Merge pull request #11 from sergey-kouzma/feat/mesto-update

gennady-bars commented on src/blocks/profile/__avatar-img/profile__avatar-img.css:
```diff
@@ -2,4 +2,5 @@
     width: 120px;
```
почему-то нет доступа к коду `Api.js`. Не весь код у Вас тут





```js
.then(this._getResponseData);

```



**Можно лучше** 

Можно сделать **универсальный метод запроса с проверкой ответа**, чтобы не дублировать эту проверку в каждом запросе:

```js

  _request(url, options) {
    return fetch(url, options).then(this._getResponseData)
  }

```

И теперь просто нужно заменить все `fetch` на `this._request`, убрав дублирование проверки на `ok`. Все остальное будет без изменений

gennady-bars commented on src/components/avatar.js:
```diff
@@ -1,40 +1,27 @@
-const profileAvatarBlock = document.querySelector('.profile__avatar');
-const profileAvatar = document.querySelector('.profile__avatar-img');
 
+function initAvatarWork(userInfo) {
```
**Можно лучше**


не нужно разбивать код по файлам. Вся инициализация классов должна быть в `index.js` только. Не нужно прятать код

gennady-bars commented on src/components/avatar.js:
```diff
@@ -1,40 +1,27 @@
-            profileAvatar.setAttribute('src', fieldAvatar.value);
-            closePopup(popupAvatarEdit);
+    function saveAvatar(data) {
+        (new Api(apiConfig)).updateAvatarAtServer(data.img)
```
`new Api(apiConfig)` не нужно создавать постоянно экземпляры `Api`. Это делается 1 раз, а потом экземпляр апи импортируется туда, где нужно

gennady-bars commented on src/components/card.js:
```diff
@@ -1,136 +1,119 @@
-// const cardTemplate = document.querySelector("#card-template").content;
-const cardTemplate = document.querySelector("#card-template").content.querySelector(".card");
-const popupWithBigImage = document.querySelector('.popup_big-image');
-const bigImg = popupWithBigImage.querySelector(".full-picture__img");
-const bigImgText = popupWithBigImage.querySelector(".full-picture__text");
+import PopupWithImage from "./PopupWithImage.js";
+import Section from './Section';
 
```
```js


import { Api } from "./Api.js";
import { apiConfig } from "./consts/api-consts.js";
import PopupWithImage from "./PopupWithImage.js";
import Section from './Section';

const cardsContainer = '.elements';

```


все это нужно удалить


По заданию **нельзя** использовать снаружи класса или импортировать в файлы с классами что-либо (только класс `Popup` можно для наследования). Нужно использовать только **мягкое связывание**

gennady-bars commented on src/components/card.js:
```diff
@@ -1,136 +1,119 @@
+
+  _getElement() {
+    const cardElement = document
+      .querySelector("#card-template")
```
`"#card-template"` селектор темплейта по заданию нужно передавать в конструктор класса `Card` из `index.js` при создании карточки

gennady-bars commented on src/components/card.js:
```diff
@@ -1,136 +1,119 @@
+    // bigImg.setAttribute("src", this._link);
+    // bigImg.setAttribute("alt", this._title);
+    // bigImgText.textContent = this._title;
+    (new PopupWithImage('.popup_big-image')).open(this._title, this._link);
```
Каждый попап нужно создать только **1 раз**  в теле файла и вызвать у него 1 раз `setEventListeners`, так как попапы всегда находятся в `DOM` и достаточно 1 раз навесить все обработчики на них.

gennady-bars commented on src/components/card.js:
```diff
@@ -1,136 +1,119 @@
     this._cardImg = this._cardElement.querySelector(".card__img");
+    this._cardImg.src = this._link;
+    this._cardImg.alt = this._title;
     this._cardLikesAmount = this._cardElement.querySelector(".card__likes-amount");
```
Отлично, что нашли элементы `DOM` 1 раз тут и сделали их полями класса.

Поля классов (так называемые **классовые** переменные с `this`) доступны везде по коду класса и внутри любого метода, особенно, если это поле класса было создано в **конструкторе**. Поэтому не нужно их передавать в вызовы методов этого же класса, так как мы можем внутри метода использовать эту переменную с `this`.

gennady-bars commented on src/components/card.js:
```diff
@@ -1,136 +1,119 @@
-
+    this._myLikeChecker()
+    this._myCardChecker();
     this._addLikesListener();
     this._addRemoveListener();
+    this._cardImg.addEventListener("click", () => this._openBigImage());
```
```js

    this._addLikesListener();
    this._addRemoveListener();
    this._cardImg.addEventListener("click", () => this._openBigImage());
```


все 3 обработчика карточки нужно навешивать в отдельном методе `_setEventListeners`, чтобы разделить логику

gennady-bars commented on src/components/card.js:
```diff
@@ -1,136 +1,119 @@
-      isOwnCard: currentUserId === card.owner._id
-    }).addCardToSite();
-  });
+function setInitialCards(data, userId) {
+  const cardList = new Section({
+    renderer: item => {
+      const card = new Card(item, userId);
+      const cardElement = card.createCard();
+      cardList.addItem(cardElement);
+    }
+  }, cardsContainer)
+  cardList.renderItems(data.reverse());
+  return cardList;
```
не нужно прятать код в файлах, в которых должен быть только объявлен класс

gennady-bars commented on src/components/card.js:
```diff
@@ -1,136 +1,119 @@
-    }).addCardToSite();
-  });
+function setInitialCards(data, userId) {
+  const cardList = new Section({
```
Экземпляр класса `Section` нужно создать только 1 раз в теле файла, чтобы он один добавлял новые карточки в `DOM` и отрисовывал начальный массив. Секция карточек у Вас **одна** на сайте

gennady-bars commented on src/components/card.js:
```diff
@@ -1,136 +1,119 @@
+function setInitialCards(data, userId) {
+  const cardList = new Section({
+    renderer: item => {
+      const card = new Card(item, userId);
```
Для создания новой карточки нужно сделать отдельную функцию `createCard` , чтобы не дублировать код и разделить логику вставки: она будет возвращать готовую карточку с уже установленными обработчиками через `return`, а вставлять в DOM там **не нужно**.
Это нужно для того, чтобы можно было разделить логику вставки и спрятать логику создания (в 5-6 спринтах карточка создается с помощью клонирования, потом (7-9) с помощью класса `Card`, а потом еще как-то в будущем). Можно же вставлять карточку в начало с `prepend`, в конец с `append`, вообще не вставлять, а сделать массив готовых карточек, а потом уже вставить всё разом в DOM. И такой пункт есть в чек-листе. Скрин из него http://joxi.ru/a2XNb9QtQj3Eer 

На всякий случай повторюсь: в функции `createCard` вставлять карточку в DOM  **не нужно**.

```js		

function createCard(item) {
    // тут создаете карточку и возвращаете ее
  return cardElement
}

```

тут останется 2 строчки кода:

```js
    renderer: item => {
      const cardElement = createCard(item)
      cardList.addItem(cardElement);
    }


```

gennady-bars commented on src/components/place.js:
```diff
@@ -1,54 +1,33 @@
-            
-            formAddPlace.reset();
-            closePopup(popupAddPlace);
+            (new Section({}, cardsContainer).addItem(new Card(card, card.owner._id).createCard()))
```
Для создания новой карточки нужно сделать отдельную функцию `createCard` , чтобы не дублировать код и разделить логику вставки

gennady-bars replied:
* Экземпляр класса `Section` нужно создать только 1 раз в теле файла, чтобы он один добавлял новые карточки в `DOM` и отрисовывал начальный массив. Секция карточек у Вас **одна** на сайте

gennady-bars commented on src/components/FormValidation.js:
```diff
@@ -0,0 +1,81 @@
+        this._inactiveButtonSelector = config.inactiveButtonSelector;
+        this._inputBlockSelector = config.inputBlockSelector;
+        this._formElement = anyPopup;
+        this._inputList = Array.from(this._formElement.querySelectorAll(this._config.inputSelector));
```
Отлично, что нашли элементы `DOM` 1 раз тут и сделали их полями класса.

Поля классов (так называемые **классовые** переменные с `this`) доступны везде по коду класса и внутри любого метода, особенно, если это поле класса было создано в **конструкторе**. Поэтому не нужно их передавать в вызовы методов этого же класса, так как мы можем внутри метода использовать эту переменную с `this`.

gennady-bars commented on src/components/Popup.js:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    _setEventListeners() {
+        document.addEventListener('keyup', (e) => this._closeByEsc(e));
```
`        document.addEventListener('keyup', (e) => this._closeByEsc(e));
`

это нужно удалить из `_setEventListeners`


По заданию и чек-листу обработчик Escape должен правильно удаляться при закрытии попапа

и в таком варианте невозможно удалить обработчик, так как Вы его **ссылку** не передаете. Там Вы передаете **анонимную** стрелочную функцию (у неё нет постоянной ссылки). Движок браузера не сможет найти по ссылке ее и удалить. Ошибки никакой браузер не выдаст, так как ему без разницы, что он не нашел функцию.


Для того чтобы ссылка стала постоянной, нужно "привязать" функцию к контексту `this` в конструкторе 1 раз с помощью `bind`.

```js
 constructor(popupSelector) {
    this._popup = …………….
    this._handleEscClose = this._handleEscClose.bind(this) <==== привязали 1 раз ===
  }


```

и теперь при открытии навешиваете и передаете только `ссылку`:


```js
 open() {
    открыли попап
    document.addEventListener('keydown', this._handleEscClose); <=== только cсылку передаете ==
    
  }

```

и удаляете при закрытии точно с такой же ссылкой:

```js
 close() {
    закрыли попап
    document.removeEventListener('keydown', this._handleEscClose); <==== такая же ссылка должна быть
  }


```

Подробнее про правильное удаление обработчиков тут: https://learn.javascript.ru/introduction-browser-events#addeventlistener

gennady-bars replied:
`        document.addEventListener('keyup', (e) => this._closeByEsc(e));
`

Это забыли удалить из `_setEventListeners`


Все делается в `open`, а удаляется в `close`

gennady-bars commented on src/components/Popup.js:
```diff
@@ -0,0 +1,37 @@
+    
+    _closeByOverlay(event) {
+        if (event.target === event.currentTarget) {
+            this.close(event.target);
```
`event.target` тут не нужно передавать в вызов

gennady-bars commented on src/components/PopupWithForm.js:
```diff
@@ -0,0 +1,47 @@
+  }
+
+  _getInputValues() {
+    this._inputList = this._form.querySelectorAll('.form__field ');
```
Все элементы, которые используются в нескольких разных методах классов или поиск которых осуществляется при каждом срабатывании метода, желательно найти **1 раз** и объявить их полями класса (**классовыми** переменными с `this`). Чаще всего это удобно сделать в `конструкторе`, `generateCard` или `_setEventListeners`, и тогда эти переменные будут **доступны** по всему коду класса. Это позволит сэкономить время на повторный поиск элементов, так как они 1 раз были найдены. И не надо будет их больше передавать в вызовы методов. И в **портфолио** такой код будет выгодно смотреться для работодателя.
Эти элементы обычно `this._inputList`, `this._submitButton`, `this._form`, `this._cardImage`, `this._likeButton`

gennady-bars replied:
вот это не исправил, я имел ввиду

gennady-bars commented on src/components/PopupWithForm.js:
```diff
@@ -0,0 +1,47 @@
+  }
+
+  resetLoadingStatus() {
+    this._submitButton.textContent = 'Сохранить';
```
**Можно лучше**

Лучше сделать метод `renderLoading` в классе `PopupWithForm`. Там найти 1 раз кнопку сабмита в конструкторе и менять ей текст.

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // фиксируем начальный текст кнопки 1 раз в конструкторе
    this._submitBtnText = this._submitBtn.textContent
  }
  // указываем 2 параметра (2й с текстом по умолчанию, чтобы не указывать лишний раз его)
  renderLoading(isLoading, loadingText='Сохранение...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```

И теперь не нужно даже думать, как возвращать начальный текст кнопке. Все будет работать для любого текста

gennady-bars commented on src/components/PopupWithForm.js:
```diff
@@ -0,0 +1,47 @@
+      this._handleFormSubmit(this._getInputValues());
+    });
+  }
+
```
**Можно лучше**

Можно сделать метод ` setInputValues` в классе `PopupWithForm`, который будет вставлять данные в инпуты:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // тут вставляем в `value` инпута данные из объекта по атрибуту `name` этого инпута
      input.value = data[input.name];
    });
  }


```

И не нужно будет искать эти инпуты в `index.js` и что-то вставлять в них при открытии профиля.

gennady-bars commented on src/components/profile.js:
```diff
@@ -1,60 +1,38 @@
-    formProfileEdit.addEventListener("submit", function (event) {
-        event.preventDefault();
+    function setFieldsToEditProfileForm() {
+        fieldProfileName.value = userInfo.getUserInfo().userName;
```
`userInfo.getUserInfo()` не нужно 2 раза вызывать 


```js

        const {userName, userInfo} = userInfo.getUserInfo()
        fieldProfileName.value = userName;
        fieldProfileDescription.value = userInfo;

```

gennady-bars commented on src/components/UserInfo.js:
```diff
@@ -0,0 +1,21 @@
+          };
+    }
+
+    setUserInfo(profileData) {
```
**Можно лучше**


С сервера **всегда** приходят все 4 вида данных пользователя при любых изменениях профиля: имя, профессия, аватар и `_id`. Вот все эти 4 вида данных можно было устанавливать в классе `UserInfo` в одном методе, **деструктурируя** ответ от сервера: `setUserInfo( { name, about, avatar, _id } )`

gennady-bars commented on src/components/index.js:
```diff
@@ -1,41 +1,34 @@
+
+const userInfo = new UserInfo(profileName, profileDescription, profileAvatar);
 
 const api = new Api(apiConfig);
```
вот этот экземпляр нужно везде использовать, а не создавать постоянно

gennady-bars commented on src/components/index.js:
```diff
@@ -1,41 +1,34 @@
-    submitButtonSelector: '.form__button',
-    inactiveButtonSelector: 'form__button_disactive',
-    errorClass: '.form__field-error'
+api.getProfileInfoFromServer(),
+api.getInitialCards()])
+.then(([profileData, initialCards]) => {    //попадаем сюда, когда оба промиса будут выполнены, деструктурируем ответ
+    userInfo.setUserInfo(profileData);
+    const cardList = setInitialCards(initialCards, profileData._id);
+    initPlacesAdding(cardList);
+})
+.catch((err) => {       //попадаем сюда если один из промисов завершится ошибкой 
+    console.log(err);
 });
```
Вам нужно **отформатировать** код, чтобы были видны нормальные отступы и вложенность одних компонентов в другие. И `js`-код тоже будет удобно читать. Это можно сделать комбинацией клавиш `SHIFT + ALT + F` или с помощью `prettier`

```jsx

Promise.all([api.getProfileInfoFromServer(), api.getInitialCards()])
  .then(([profileData, initialCards]) => {
    userInfo.setUserInfo(profileData);
    const cardList = setInitialCards(initialCards, profileData._id);
    initPlacesAdding(cardList);
  })
  .catch((err) => {
    console.log(err);
  });

```

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на **гитхабе** все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная, но есть некоторые недочеты: 




* при редактировании аватара в консоли ошибка https://disk.yandex.ru/i/3DhNdxwOOsfB5g 
* файлы `index.js` и `index.css` по чек-листу должны быть в папке `pages`
* Все файлы с классами по чек-листу нужно сделать с **большой** буквы. 
* Файлы `constants.js` и `utils.js` нужно поместить в папку `utils`, так как в них нет компонентов или классов.
* `new Api(apiConfig)` не нужно создавать постоянно экземпляры `Api`. Это делается 1 раз, а потом экземпляр апи импортируется туда, где нужно
* По заданию **нельзя** использовать снаружи класса или импортировать в файлы с классами что-либо (только класс `Popup` можно для наследования). Нужно использовать только **мягкое связывание**
* `"#card-template"` селектор темплейта по заданию нужно передавать в конструктор класса `Card` из `index.js` при создании карточки
* Каждый попап нужно создать только **1 раз**  в теле файла и вызвать у него 1 раз `setEventListeners`, так как попапы всегда находятся в `DOM` и достаточно 1 раз навесить все обработчики на них.
* все 3 обработчика карточки нужно навешивать в отдельном методе `_setEventListeners`, чтобы разделить логику
* Экземпляр класса `Section` нужно создать только 1 раз в теле файла, чтобы он один добавлял новые карточки в `DOM` и отрисовывал начальный массив. Секция карточек у Вас **одна** на сайте
* Для создания новой карточки нужно сделать отдельную функцию `createCard` , чтобы не дублировать код и разделить логику вставки
* По заданию и чек-листу обработчик Escape должен правильно удаляться при закрытии попапа
* Все элементы, которые используются в нескольких разных методах классов или поиск которых осуществляется при каждом срабатывании метода, желательно найти **1 раз** и объявить их полями класса



Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте **работоспособность проекта** и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

Vasyankin-Alexander-Henchfiend committed:
Иправления после ревью ч1

Исправлено:
- при редактировании аватара в консоли ошибка https://disk.yandex.ru/i/3DhNdxwOOsfB5g
- файлы index.js и index.css по чек-листу должны быть в папке pages
- Все файлы с классами по чек-листу нужно сделать с большой буквы.
- Файлы constants.js и utils.js нужно поместить в папку utils, так как в них нет компонентов или классов.
- все 3 обработчика карточки нужно навешивать в отдельном методе _setEventListeners, чтобы разделить логику

Vasyankin-Alexander-Henchfiend committed:
Иправления после ревью ч2

- По заданию и чек-листу обработчик Escape должен правильно удаляться при закрытии попапа
- "#card-template" селектор темплейта по заданию нужно передавать в конструктор класса Card из index.js при создании карточки

Vasyankin-Alexander-Henchfiend committed:
Исправления после ревью ч3
- переработан вызов классов Card и Section теперь они вызываются только один раз
- переработано получение данных для поля ввода в форму редактирования информации о пользователе
- несколько констант перенесены из файлов классов в файл для констант

Vasyankin-Alexander-Henchfiend committed:
asds

Vasyankin-Alexander-Henchfiend committed:
sdfdsf

Vasyankin-Alexander-Henchfiend committed:
asxs

Vasyankin-Alexander-Henchfiend committed:
adsasd

to-review committed:
рефакторинг класса карточек и api

sergey-kouzma committed:
Merge pull request #13 from sergey-kouzma/feat/mesto-update-card-refactoring

рефакторинг класса карточек и api

gennady-bars reviewed (commented):
Уже почти все отлично, но:


* кое-что не исправили https://github.com/sergey-kouzma/mesto-project/pull/10#discussion_r1163362238 
* и это тоже https://github.com/sergey-kouzma/mesto-project/pull/10/files/a45a546c868ade46332096c07a9cf74486097e56..c55b786ba0ccf8cdaa944f1f2d84792220220ca5#r1163361650 .

to-review committed:
review fixes

sergey-kouzma committed:
Merge pull request #14 from sergey-kouzma/feat/mesto-update-last-fixes

review fixes

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
